modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogPostAdminAppService.cs

```diff
@@ -110,4 +113,30 @@ public virtual async Task DeleteAsync(Guid id)
     {
         await BlogPostRepository.DeleteAsync(id);
     }
-}
+
+    [Authorize(CmsKitAdminPermissions.BlogPosts.Publish)]
+    public virtual async Task PublishAsync(Guid id)
+    {
+        var blogPost = await BlogPostRepository.GetAsync(id);
+        blogPost.SetPublished();
+    }
+
+    [Authorize(CmsKitAdminPermissions.BlogPosts.Update)]
+    public virtual async Task DraftAsync(Guid id)
+    {
+        var blogPost = await BlogPostRepository.GetAsync(id);
+        blogPost.SetDraft();
+    }
```

Comment:
Calling update method of repository here might be better since MongoDB doesn't have entity tracking system and it won't be updated in MongoDB at the end of uow. In other hand, transaction can be disabled for MongoDB, in that case, this entity won't be updated
 
```suggestion
        blogPost.SetDraft();
        await BlogPostRepository.UpdateAsync(blogPost);
    }
```

Reply:
seems also need same changes in PublishAsync method

Reply:
Yeah, same for **PublishAsync()** method too

Reply:
Thank you. It is resolved in [03e7905](https://github.com/abpframework/abp/pull/12073/commits/03e790547fbc0fb02589edb216218b657d9367d5)